pilot/pkg/model/context.go

```diff
@@ -151,6 +152,35 @@ func (node *Proxy) ServiceNode() string {
 
 }
 
+// Isolated return true if the node should only return services from its own namespace
+// or explicit imports.
+func (node *Proxy) Isolated() bool {
+	if node == nil {
+		return false
+	}
+
+	// Gateways use explicit imports (or explicit bind to gateway)
+	if node.Type != Sidecar {
+		return false
+	}
+
+	// None interception mode requires strict isolation, since it depends on port uniqueness for TCP services.
+	if node.Metadata[pilot.InterceptionMode] == pilot.InterceptionModeNone {
+		return true
+	}
+
+	if node.Metadata[pilot.Isolation] != "" {
+		return true
+	}
+
+	// Global enable flag, containing default namespaces
+	if pilot.NetworkScopes != "" {
+		return true
```

Comment:
per the new API design, we no longer have any global flag with default namespaces.. The plan is to let people specify these in the Sidecar resource for every namespace. Secondly, we also dont have any global flag of any sort for local-namespace-only by default. Its all through the sidecar resource. Its verbose but for the moment, this should suffice. Otherwise we will have very confusing results when you define sidecar resource and also have global flags